model/task/db.go

```diff
@@ -802,6 +802,105 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+type TaskBuildVariants struct {
+	Task          string   `bson:"_id"`
+	BuildVariants []string `bson:"build_variants"`
+}
+
+// FindUniqueBuildVariantNamesByTask returns TaskBuildVariants with a list of unique build variants
+func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) (*TaskBuildVariants, error) {
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			ProjectKey:     projectId,
+			DisplayNameKey: taskName,
+			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes}},
+		}}
+
+	group := bson.M{
+		"$group": bson.M{
+			"_id":                 taskName,
+			BuildVariantKey + "s": bson.M{"$addToSet": "$" + BuildVariantKey},
```

Comment:
is this key the "build_variants" field referenced at 827 or something else? For the purposes of clarify I think it may actually be better to hardcode this (same at 837)